capstone/src/main/java/com/google/univiz/api/SearchResourceImpl.java

```diff
@@ -0,0 +1,31 @@
+package com.google.univiz.api;
+
+import com.google.common.collect.Lists;
+import com.google.univiz.api.representation.CollegeId;
+import com.google.univiz.api.representation.SearchData;
+import com.google.univiz.api.representation.SuggestionResponse;
+import com.google.univiz.api.resource.SearchResource;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+
+final class SearchResourceImpl implements SearchResource {
+  private final SuggestionDataApi suggestionApi;
+
+  @Inject
+  public SearchResourceImpl(SuggestionDataApi suggestionApi) {
+    this.suggestionApi = suggestionApi;
+  }
+
+  @Override
+  public List<SearchData> getSearchSuggestions(String partialCollegeName) {
+    if (partialCollegeName == null || !partialCollegeName.matches("[a-zA-Z]+")) {
+      return Lists.newArrayList(SearchData.create("", CollegeId.create(0)));
```

Comment:
How about returning an empty list instead of a list with a single dummy element in it. That's a clear signal to the client that there weren't any results, instead of requiring the client to inspect the list for a dummy result.

Reply:
fixed!